option java_package="com.baidu.soushang.lbs";
option java_outer_classname="Models";
option optimize_for = LITE_RUNTIME;

message User {
    required uint64 id = 1;
    required string name = 2;
    required string avatar = 3;
    required int32 net_type = 4;
    required int32 fight_num = 5;
    required int32 win_num = 6;
    required int32 state = 7;
}

message CommandMsg {
    required int32 type = 1;
    required bytes content = 2;
}

message IClientInfo {
    required uint64 id = 1;
    required string name = 2;
    required string avatar = 3;
    required int32 net_type = 4;
}

message IClientLBS {
    required float longitude = 1;
    required float latitude = 2;
}

message OQuestion {
    required string fight_key = 1;
}
message IAnswer {
    required int32 index = 1;
    required int32 right = 2;
}
message OFightState {
    required int32 all = 1;
    required int32 done = 2;
    required int32 right = 3;
}
message OFightResult {
    // 1 is win 2 is lose
    // 3 is other lose connect, you win
    required int32 result = 1;
    required float me_win_ratio = 2;
    required int32 me_score = 3;
    required int32 me_point = 4;
    required int32 other_point = 5;
    required int32 other_time = 6;
    required int32 me_time = 7;
}

message OPeerListResp {
    repeated User users = 1;
}

message IFightReq {
    required uint64 id = 1;
    required int32 bet = 2;
}

message OFightResp {
    // 0 shows success
    // 1 shows failed
    // 2 shows bet too much
    required int32 result = 1; 
    optional string message = 2;
}
message OFightReq {
    required User user = 1;
    required int32 bet = 2;
}
message IFightResp {
    // 0 is agree, others disagree
    required int32 result = 1;
}

message EmptyMsg {
}
